[#168141240] Fix bug if sender is unknown

valentinacala opened the pull request:
This pr solves the bug causing error screen if the content  of a message related service is not available (into `state.entities.service.byId`).
If `service_id` is undefined and/or the servce is none, the message detail des not displayes the logo, the organization name and the service name (the latters being ""unkown sender/service)

Here  an example after the implementation:
![messageNoServiceAvailable](https://user-images.githubusercontent.com/38431762/63939214-02dd2380-ca67-11e9-8dcc-70ec7adf0575.gif)

digitalcitizenship commented:
<!--
  0 failure: 
  1 warning:  Please include a ...
  
  1 markdown notices
  DangerID: danger-id-default;
-->


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Please include a Pivotal story at the beginning of the PR title (see below).
  </td>
    </tr>
  </tbody>
</table>



  Example of PR titles that include pivotal stories:

  * single story: `[#123456] my PR title`
  * multiple stories: `[#123456,#123457,#123458] my PR title`

    
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

Undermaken commented on ts/components/messages/MessageDetailComponent.tsx:
```diff
@@ -121,24 +121,26 @@ export default class MessageDetailComponent extends React.PureComponent<Props> {
+                <Col>
+                  <H4>{service.organization_name}</H4>
+                  <H6 link={true} onPress={onServiceLinkPress}>
+                    {service.service_name}
```
Adding **_pot.isSome(potService)_** nothing will be show if service is unknown (_pot.none_)
Could be a better approach if we add **_service_name_** here
https://github.com/teamdigitale/io-app/blob/48a27ac1cb5a0675935770cd1fab961bf164af61/ts/components/messages/MessageDetailComponent.tsx#L105 
and then render the logo only if **_service_id_** is defined?

valentinacala replied:
From the service definition I read  `service_id` is available only if `potService` is `some`. Is it a wrong reading of the service definition?

Undermaken replied:
Yes of course, **_service_id_** is available only if pot is some.
Anyway we should show info about the service is unknown.

**This is how a full loaded message looks like**
<img src="https://user-images.githubusercontent.com/822471/64028980-fbd81300-cb43-11e9-8909-5d5156acb36d.png" height="500"/>

With this approach no info about service is shown
<img src="https://user-images.githubusercontent.com/822471/64028513-13fb6280-cb43-11e9-9d9d-9a5e13fd3e9f.png" height="500"/>

Changing the code in this way

```typescript
const service =
      potService !== undefined
        ? pot.isNone(potService)
          ? ({
              organization_name: I18n.t("messages.errorLoading.senderService"),
              department_name: I18n.t("messages.errorLoading.senderInfo"),
              service_name: ""
            } as ServicePublic)
          : pot.toUndefined(potService)
        : undefined;

    const payment =
      message.content.payment_data !== undefined &&
      service !== undefined &&
      service.organization_fiscal_code
        ? paymentsByRptId[
            `${service.organization_fiscal_code}${
              message.content.payment_data.notice_number
            }`
          ]
        : undefined;

    return (
      <View style={styles.mainWrapper}>
        <View style={styles.headerContainer}>
          {/* Service */}
          {service && (
            <Grid style={styles.serviceContainer}>
              <Col>
                <H4>{service.organization_name}</H4>
                <H6 link={true} onPress={onServiceLinkPress}>
                  {service.service_name}
                </H6>
              </Col>
              {service.service_id && (
                <Col style={styles.serviceCol}>
                  <TouchableOpacity onPress={onServiceLinkPress}>
                    <MultiImage
                      style={styles.serviceMultiImage}
                      source={logosForService(service)}
                    />
                  </TouchableOpacity>
                </Col>
              )}
            </Grid>
          )}
```
service details will be shown
<img src="https://user-images.githubusercontent.com/822471/64028841-ad2a7900-cb43-11e9-81fd-a40bcb34c034.png" height="500"/>

I think it's preferred to show a message instead of nothing

Undermaken commented:
@valentinacala Sorry, could you also include in this PR the removing of this
https://github.com/teamdigitale/io-app/blob/b355d55866af25e4f376d4573137e383eb1a79de/ts/components/messages/MessageListItem.tsx#L114-L118
from there to **_locales_**?

valentinacala commented:
> @valentinacala Sorry, could you also include in this PR the removing of this
> 
> https://github.com/teamdigitale/io-app/blob/b355d55866af25e4f376d4573137e383eb1a79de/ts/components/messages/MessageListItem.tsx#L114-L118
> 
> from there to **_locales_**?

updated within https://github.com/teamdigitale/io-app/pull/1279/commits/8499ac2f60c534d16434e8125a8ac2f21dd2d708

valentinacala commented:
Within the last three commits (https://github.com/teamdigitale/io-app/pull/1279/commits/b57f1383108c7e30c3a646d726d0c63f70e36b6c, https://github.com/teamdigitale/io-app/pull/1279/commits/086ae3dd4e35b911679e3559b78f74b27d184d8f and https://github.com/teamdigitale/io-app/pull/1279/commits/c6134919773ebaee776e66c4279b63fd65189230) I updated the code as sugegsted. I added also:
- service_name to the mocked unknown `service` to make 'Service info unknown' being displayed under 'Sender service unknown'
- a locale string to display 'Department info unknown' under the message details

Here the result: 
<img width="265" alt="image" src="https://user-images.githubusercontent.com/38431762/64034490-106dd880-cb4f-11e9-9c88-cd00d162299c.png">
